### PR TITLE
Add treasury donation to tx table

### LINKED ISSUE #1732: Treasury donations
With Conway it's possible to send money to the treasury. We should capture this in a new field in `tx` which is by default and for previous entries 0. Also we should take this into account in out tx fee calculation.

### PR DESCRIPTION
**Pull request by sgillespie**
# Description

Fixes #1732 

# Checklist

- [x] Commit sequence broadly makes sense
- [x] Commits have useful messages
- [x] New tests are added if needed and existing tests are updated
- [x] Any changes are noted in the [changelog](https://github.com/IntersectMBO/cardano-db-sync/blob/master/cardano-db-sync/CHANGELOG.md)
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) on version 0.10.1.0 (which can be run with `scripts/fourmolize.sh`)
- [x] Self-reviewed the diff

# Migrations

- [x] The pr causes a [breaking change](https://github.com/IntersectMBO/cardano-db-sync/blob/master/doc/migrations.md) of type a,b or c
- [x] If there is a breaking change, the pr includes a database migration and/or a fix process for old values, so that upgrade is possible
- [x] Resyncing and running the migrations provided will result in the same database semantically

If there is a breaking change, especially a big one, please add a justification here. Please elaborate
more what the migration achieves, what it cannot achieve or why a migration is not possible.

### CODE REVIEW

**Comment by Cmdv** [comment]
looks like theres failures when building the tests 😞

**Review by kderme — APPROVED** [review]
Just one comment, seems good otherwise

File: schema/migration-2-0044-20240626.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
+-- Persistent generated migration.
```
[the remark this answers is not in the document]
Comment (sgillespie) [reply schema/migration-2-0044-20240626.sql]: Done!

**Comment by sgillespie** [comment]
> looks like theres failures when building the tests 😞

Fixed!

**Review by kderme — APPROVED** [review]
LGTM! I left a comment about some missing functionality. We could merge this and do it on a separate pr though.

File: cardano-db-sync/src/Cardano/DbSync/Era/Universal/Insert/Tx.hs [hunk 2]
```diff
@@ -77,6 +77,7 @@ insertTx syncEnv isMember blkId epochNo slotNo applyResult blockIndex tx grouped
   let !mdeposits = if not (Generic.txValidContract tx) then Just (Coin 0) else lookupDepositsMap txHash (apDepositsMap applyResult)
   let !outSum = fromIntegral $ unCoin $ Generic.txOutSum tx
       !withdrawalSum = fromIntegral $ unCoin $ Generic.txWithdrawalSum tx
+      !treasuryDonation = fromIntegral . unCoin $ Generic.txTreasuryDonation tx
```
Comment (kderme) [inline cardano-db-sync/src/Cardano/DbSync/Era/Universal/Insert/Tx.hs:80]: IIrc we wanted to use this in the deposit calculation for a specifc case.

Comment (sgillespie) [reply cardano-db-sync/src/Cardano/DbSync/Era/Universal/Insert/Tx.hs]: Thanks for the reminder, I just updated it to be subtracted from the inputs. However, I couldn't find a way to test it, so please carefully re-review

**Review by kderme — APPROVED** [review]
LGTM!

**Pull request merged**